expression/builtin_cast.go

```diff
@@ -461,9 +461,10 @@ func (b *builtinCastIntAsRealSig) evalReal(row chunk.Row) (res float64, isNull b
 	if isNull || err != nil {
 		return res, isNull, err
 	}
-	if !mysql.HasUnsignedFlag(b.tp.Flag) && !mysql.HasUnsignedFlag(b.args[0].GetType().Flag) {
+	unsigned := mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
```

Comment:
It's better to define the `unsigned` in the `IF` statement to restrict its scope. e.g `if unsigned := ...; unsigned... {}`